sphinx_revealjs/writers.py

```diff
@@ -86,7 +86,19 @@ def visit_literal_block(self, node: literal_block):
         Override base method, and open simply ``pre`` and ``code`` tags.
         """
         lang = node["language"]
-        self.body.append(f'<pre><code data-trim data-noescape class="{lang}">\n')
+        self.body.append("<pre")
+        # add section id as data-id
+        if isinstance(node.parent, section) and len(node.parent["ids"]):
+            self.body.append(f" data-id=\"{node.parent['ids'][0]}\"")
+        self.body.append(f'><code data-trim data-noescape class="{lang}"')
+        # use the emphasize-lines directive to create line for line animations
+        if "highlight_args" in node and "hl_lines" in node["highlight_args"]:
+            self.body.append(f"data-line-numbers=\"{'|'.join(map(str,node['highlight_args']['hl_lines']))}\"")
```

Comment:
`emphasize-lines` option are converted from syntax of option to list of numbers in Sphinx
(ex: `1-3,5` is converted to `[1,2,3,5]` ).

This implementation highlight single line only in order (`[1,2,3,5]`=>`1|2|3|5`).
It does not match expection by user that want to higilight multi-lines.

Reply:
Thanks for your feedback. Yeah, the solution is far from ideal, it's a bit of hackery.
Personally I think that overriding the code-block directive is the cleanest solution, adding the 'extra' attributes like `line-numbers` and `data-id`. But this needs some work... :flushed: